packages/grant-explorer/src/features/round/ViewRoundPage.tsx

```diff
@@ -352,19 +342,25 @@ function ProjectCard(props: {
   roundRoutePath: string;
   isBeforeRoundEndDate?: boolean;
   roundId: string;
+  chainId: ChainId;
+  setCurrentProjectAddedToCart: React.Dispatch<React.SetStateAction<Project>>;
+  setShowCartNotification: React.Dispatch<React.SetStateAction<boolean>>;
 }) {
   const { project, roundRoutePath } = props;
   const projectRecipient =
     project.recipient.slice(0, 5) + "..." + project.recipient.slice(-4);
 
-  const [cart, handleAddProjectsToCart, handleRemoveProjectsFromCart] =
-    useCart();
+  const { projects, add, remove } = useCartStorage();
 
-  const isAlreadyInCart = cart.some(
+  const isAlreadyInCart = projects.some(
     (cartProject) =>
       cartProject.grantApplicationId === project.grantApplicationId
   );
 
+  const cartProject = project as CartProject;
```

Comment:
Why the cast? Aren't `Project` and `CartProject` incompatible?

Reply:
They are incompatible, but it's then modified on the next two lines. OOS, not the OP of that code, but will fix.